### PR TITLE
Prevent Institutions from Being Create or Modified to Include a public Email Domain

### PR DESCRIPTION
**Pull request by kgudel**
Adds:
1. A list of public email domains in common
2. Checks the above list when an institution is created or modified to prevent inclusion of public domains in institution records
3. Prevents any results being returned when a public email domain is search in the institutions-api

### CODE REVIEW

File: common/src/main/resources/reference.conf [hunk 1]
```diff
@@ -16,6 +16,11 @@ hmda {
...
+    publicDomains = ${?PUBLIC_EMAIL_DOMAINS}
```
Comment (PatrickGoRaft) [inline common/src/main/resources/reference.conf:21]: @kgudel  we will need to define a configmap for this environment variable

File: common/src/main/scala/hmda/api/http/EmailUtils.scala [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package hmda.api.http
+
+import com.typesafe.config.ConfigFactory
+
+// $COVERAGE-OFF$
+object EmailUtils {
+  val config = ConfigFactory.load("reference.conf")
+  val publicDomains = config.getString("hmda.email.publicDomains").split(",")
+
```
Comment (PatrickGoRaft) [inline common/src/main/scala/hmda/api/http/EmailUtils.scala:9]: @kgudel  the checks here will need to be case insensitive.

File: institutions-api/src/main/scala/hmda/institution/api/http/InstitutionQueryHttpApi.scala [hunk 3]
```diff
@@ -128,8 +129,12 @@ private class InstitutionQueryHttpApi(config: Config)(implicit ec: ExecutionCont
   path("institutions") {
     (extractUri & get) { uri =>
       parameter('domain.as[String]) { domain =>
+        if (checkIfPublicDomain(domain)) {
```
Comment (PatrickGoRaft) [inline institutions-api/src/main/scala/hmda/institution/api/http/InstitutionQueryHttpApi.scala:132]: we should protect all endpoints looking things up by domain

**Review by PatrickGoRaft — CHANGES_REQUESTED** [review]

File: hmda/src/main/scala/hmda/api/http/admin/InstitutionAdminHttpApi.scala [hunk 4]
```diff
@@ -233,5 +236,7 @@ private class InstitutionAdminHttpApi(config: Config, sharding: ClusterSharding)
       complete((StatusCodes.BadRequest, "Incorrect lei format"))
     } else if (checkAgencyCode && !validAgencyCodeFormat(institution.agency.code)) {
       complete((StatusCodes.BadRequest, "Incorrect agency code format"))
+    } else if (checkListIfPublicDomain(institution.emailDomains)){
```
Comment (PatrickGoRaft) [inline hmda/src/main/scala/hmda/api/http/admin/InstitutionAdminHttpApi.scala:239]: should remove whitespace and and make this case insensitive

File: common/src/main/resources/reference.conf (hunk 1 above)
Comment (kgudel) [reply common/src/main/resources/reference.conf:21]: I don't think it's necessary to make this a config map at the moment as this list should not change.

**Review by PatrickGoRaft — APPROVED** [review]
LGTM.

**Pull request merged**